# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/VaryingPacking.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/VaryingPacking.cpp
--- a/gfx/angle/checkout/src/libANGLE/VaryingPacking.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/VaryingPacking.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -314,10 +314,13 @@
                 if (varying->isStruct())
                 {
                     ASSERT(!varying->isArray());
-                    for (const auto &field : varying->fields)
+                    for (GLuint fieldIndex = 0; fieldIndex < varying->fields.size(); ++fieldIndex)
                     {
+                        const sh::ShaderVariable &field = varying->fields[fieldIndex];
+
                         ASSERT(!field.isStruct() && !field.isArray());
-                        mPackedVaryings.emplace_back(field, interpolation, varying->name);
+                        mPackedVaryings.emplace_back(field, interpolation, varying->name,
+                                                     fieldIndex);
                         uniqueFullNames.insert(mPackedVaryings.back().fullName());
                     }
                 }
@@ -354,11 +357,14 @@
             }
             if (input->isStruct())
             {
-                const sh::ShaderVariable *field = FindShaderVarField(*input, tfVarying);
+                GLuint fieldIndex = 0;
+                const sh::ShaderVariable *field =
+                    FindShaderVarField(*input, tfVarying, &fieldIndex);
                 if (field != nullptr)
                 {
                     ASSERT(!field->isStruct() && !field->isArray());
-                    mPackedVaryings.emplace_back(*field, input->interpolation, input->name);
+                    mPackedVaryings.emplace_back(*field, input->interpolation, input->name,
+                                                 fieldIndex);
                     mPackedVaryings.back().vertexOnly = true;
                     mPackedVaryings.back().arrayIndex = GL_INVALID_INDEX;
                     uniqueFullNames.insert(tfVarying);
@@ -425,9 +431,4 @@
 
     return true;
 }
-
-const std::vector<std::string> &VaryingPacking::getInactiveVaryingNames() const
-{
-    return mInactiveVaryingNames;
-}
 }  // namespace gl